src/community/oseo/oseo-core/src/test/java/org/geoserver/opensearch/eo/OSEOFactoryExtensionTest.java

```diff
@@ -0,0 +1,30 @@
+package org.geoserver.opensearch.eo;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import org.geoserver.catalog.WorkspaceInfo;
+import org.geoserver.config.GeoServer;
+import org.geoserver.test.GeoServerSystemTestSupport;
+import org.junit.Test;
+
+public class OSEOFactoryExtensionTest extends GeoServerSystemTestSupport {
+    @Test
+    public void testDirectCreate() throws Exception {
+        OSEOFactoryExtension oseo = new OSEOFactoryExtension();
+        OSEOInfo oseoInfo = oseo.create(OSEOInfo.class);
+        assertTrue(oseoInfo instanceof OSEOInfoImpl);
+    }
+
+    @Test
+    public void testCreateViaFactory() throws Exception {
+        GeoServer geoServer = getGeoServer();
+        WorkspaceInfo ws = geoServer.getCatalog().getWorkspaceByName("sf");
+        OSEOInfo oseoInfo = geoServer.getFactory().create(OSEOInfo.class);
+        assertNotNull(oseoInfo);
+        oseoInfo.setName("OSEO");
+        oseoInfo.setWorkspace(ws);
+        assertEquals("sf", oseoInfo.getWorkspace().getName());
```

Comment:
This... is setting a workspace and then checking its name. It's testing the test code itself? :rofl:

Reply:
removed